src/providers/ldap/ldap_id.c

```diff
@@ -292,7 +297,13 @@ struct tevent_req *users_get_send(TALLOC_CTX *memctx,
         }
     }
 
-    if (state->use_id_mapping || filter_type == BE_FILTER_SECID) {
+    if (state->non_posix == true) {
+        state->filter = talloc_asprintf(state,
```

Comment:
Is there a reason why you use explicit 'val == true' and 'val == false' instead of just 'val' and '!val'? I see that there is 'state->use_id_mapping == false' in existing code (most probably my fault) but I think in general we use the shorter variant for boolean types.